internal/types/document.go

```diff
@@ -428,6 +430,38 @@ func (d *Document) moveIDToTheFirstIndex() {
 	d.fields = slices.Delete(d.fields, idIdx+1, idIdx+2)
 }
 
+// Iterator returns an iterator over the document fields.
+func (d *Document) Iterator() iterator.Interface[string, any] {
```

Comment:
I'd move it to a separate file, smth like `document_iterator` to define a smaller context (this file is already quite long)

Reply:
Done.